packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-endpoint-config.ts

```diff
@@ -0,0 +1,135 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import * as kms from '@aws-cdk/aws-kms';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+import { ProductionVariant } from './base-types';
+import { renderTags } from './private/utils';
+
+/**
+ * Properties for creating an Amazon SageMaker endpoint configuration
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-sagemaker.html
+ * @experimental
+ */
+export interface SageMakerCreateEndpointConfigProps extends sfn.TaskStateBaseProps {
+  /**
+   * The name of the endpoint configuration.
+   */
+  readonly endpointConfigName: string;
+  /**
+   * AWS Key Management Service key that Amazon SageMaker
+   * uses to encrypt data on the storage volume attached to the ML compute instance that hosts the endpoint.
+   *
+   * @default - None
+   */
+  readonly kmsKey?: kms.IKey;
+
+  /**
+   * An list of ProductionVariant objects, one for each model that you want to host at this endpoint.
+   * Identifies a model that you want to host and the resources to deploy for hosting it.
+   * If you are deploying multiple models, tell Amazon SageMaker how to distribute traffic among the models by specifying variant weights.
+   */
+  readonly productionVariants: ProductionVariant[];
+
+  /**
+   * Tags to be applied to the endpoint configuration.
+   *
+   * @default - No tags
+   */
+  readonly tags?: { [key: string]: string };
```

Comment:
similar question as earlier, can tags be passed via state input? (i.e. referenced as `$.blah`)
if the answer to that is yes, we'll need to make the type `sfn.TaskInput` as it would need to be able to accept either an object *or* a string

note: this applies to all usages of `tags` in this PR - I think this is something we've missed elsewhere in this module as well, but no time like the present to start fixing that :)

Reply:
I really like this suggestion! Yeah, tags can be passed via state input therefore it will make sense for them to become `sfn.TaskInput`. I'll change them in the next revision

Reply:
I've changed `tags` to use `sfn.TaskInput` for the new sagemaker tasks. For the old `SageMakerCreateTrainingJob` and `SageMakerCreateTransformJob` I left them to use `{ [key: string]: string }` so that not to introduce a parameter change in this PR. I think it's better to postponed it to the next one

Reply:
awesome, that's great!